### ISSUE TITLE
Dynamic expiration

### ISSUE THREAD

**Issue by cubicbyte** [requests-cache/requests-cache#862]
Hi, is dynamic expiration time supported?

For example, I want to make a request when I have `expire_after=timedelta(days=1)`.
Then I change my mind and set it to `timedelta(minutes=5)` and make a request after 10 minutes.
I would like the previous response to expire, because it was created 10 minutes ago and the new expiration time is 5 minutes.

Is it possible to do this?

**Comment by JWCook** [comment]
Not exactly, but there are a few things you can do that are similar.

## Option 1
You can use [`reset_expiration()`](https://requests-cache.readthedocs.io/en/stable/modules/requests_cache.backends.base.html#requests_cache.backends.base.BaseCache.reset_expiration) to apply a new expiration relative to the **current time**:
```py
from datetime import timedelta
from requests_cache import CachedSession

session = CachedSession(expire_after=timedelta(days=1))
response = session.get('https://httpbin.org/json')

# Set expiration for all cached items to 5 minutes from now
session.cache.reset_expiration(timedelta(minutes=5))

response = session.get('https://httpbin.org/json')
print(response.expires)
```

## Option 2
You can also [`delete()`](https://requests-cache.readthedocs.io/en/stable/modules/requests_cache.backends.base.html#requests_cache.backends.base.BaseCache.delete) items from the cache older than a certain time:
```py
session.cache.delete(older_than=timedelta(minutes=5))
```

There is some more info in the docs under [Removing Responses](https://requests-cache.readthedocs.io/en/stable/user_guide/expiration.html#removing-responses).

## Option 3
If you need to to, you can also manually modify individual responses and save them with [`save_response()`](https://requests-cache.readthedocs.io/en/stable/modules/requests_cache.backends.base.html#requests_cache.backends.base.BaseCache.save_response):
```py
session.cache.save_response(
    response,
    expires=response.created_at + timedelta(minutes=5),
)
```

**Comment by JWCook** [comment]
Right, it's currently a backend-agnostic method that modifies responses one at a time, so it will be fairly slow for a large cache.

For the SQLite backend, it could be optimized by doing it in a single SQL statement. If that's something you need, feel free to open a new issue (or PR!) for it.
